File: web/src/api/demo_code_generation.js
import service from '@/utils/request'

// @Tags DemoCodeGeneration
// @Summary 创建DemoCodeGeneration
// @Security ApiKeyAuth
// @accept application/json
// @Produce application/json
// @Param data body model.DemoCodeGeneration true "创建DemoCodeGeneration"
// @Success 200 {string} string "{"success":true,"data":{},"msg":"获取成功"}"
// @Router /demoCodeGeneration/createDemoCodeGeneration [post]
export const createDemoCodeGeneration = (data) => {
     return service({
         url: "/demoCodeGeneration/createDemoCodeGeneration",
         method: 'post',
         data
     })
 }


// @Tags DemoCodeGeneration
// @Summary 删除DemoCodeGeneration
// @Security ApiKeyAuth
// @accept application/json
// @Produce application/json
// @Param data body model.DemoCodeGeneration true "删除DemoCodeGeneration"
// @Success 200 {string} string "{"success":true,"data":{},"msg":"删除成功"}"
// @Router /demoCodeGeneration/deleteDemoCodeGeneration [delete]
 export const deleteDemoCodeGeneration = (data) => {
     return service({
         url: "/demoCodeGeneration/deleteDemoCodeGeneration",
         method: 'delete',
         data
     })
 }

// @Tags DemoCodeGeneration
// @Summary 删除DemoCodeGeneration
// @Security ApiKeyAuth
// @accept application/json
// @Produce application/json
// @Param data body request.IdsReq true "批量删除DemoCodeGeneration"
// @Success 200 {string} string "{"success":true,"data":{},"msg":"删除成功"}"
// @Router /demoCodeGeneration/deleteDemoCodeGeneration [delete]
 export const deleteDemoCodeGenerationByIds = (data) => {
     return service({
         url: "/demoCodeGeneration/deleteDemoCodeGenerationByIds",
         method: 'delete',
         data
     })
 }

// @Tags DemoCodeGeneration
// @Summary 更新DemoCodeGeneration
// @Security ApiKeyAuth
// @accept application/json
// @Produce application/json
// @Param data body model.DemoCodeGeneration true "更新DemoCodeGeneration"
// @Success 200 {string} string "{"success":true,"data":{},"msg":"更新成功"}"
// @Router /demoCodeGeneration/updateDemoCodeGeneration [put]
 export const updateDemoCodeGeneration = (data) => {
     return service({
         url: "/demoCodeGeneration/updateDemoCodeGeneration",
         method: 'put',
         data
     })
 }


// @Tags DemoCodeGeneration
// @Summary 用id查询DemoCodeGeneration
// @Security ApiKeyAuth
// @accept application/json
// @Produce application/json
// @Param data body model.DemoCodeGeneration true "用id查询DemoCodeGeneration"
// @Success 200 {string} string "{"success":true,"data":{},"msg":"查询成功"}"
// @Router /demoCodeGeneration/findDemoCodeGeneration [get]
 export const findDemoCodeGeneration = (params) => {
     return service({
         url: "/demoCodeGeneration/findDemoCodeGeneration",
         method: 'get',
         params
     })
 }


// @Tags DemoCodeGeneration
// @Summary 分页获取DemoCodeGeneration列表
// @Security ApiKeyAuth
// @accept application/json
// @Produce application/json
// @Param data body request.PageInfo true "分页获取DemoCodeGeneration列表"
// @Success 200 {string} string "{"success":true,"data":{},"msg":"获取成功"}"
// @Router /demoCodeGeneration/getDemoCodeGenerationList [get]
 export const getDemoCodeGenerationList = (params) => {
     return service({
         url: "/demoCodeGeneration/getDemoCodeGenerationList",
         method: 'get',
         params
     })
 }

File: server/router/demo_code_generation.go
package router

import (
	"gin-vue-admin/api/v1"
	"gin-vue-admin/middleware"
	"github.com/gin-gonic/gin"
)

func InitDemoCodeGenerationRouter(Router *gin.RouterGroup) {
	DemoCodeGenerationRouter := Router.Group("demoCodeGeneration").Use(middleware.JWTAuth()).Use(middleware.CasbinHandler()).Use(middleware.OperationRecord())
	{
		DemoCodeGenerationRouter.POST("createDemoCodeGeneration", v1.CreateDemoCodeGeneration)   // 新建DemoCodeGeneration
		DemoCodeGenerationRouter.DELETE("deleteDemoCodeGeneration", v1.DeleteDemoCodeGeneration) // 删除DemoCodeGeneration
		DemoCodeGenerationRouter.DELETE("deleteDemoCodeGenerationByIds", v1.DeleteDemoCodeGenerationByIds) // 批量删除DemoCodeGeneration
		DemoCodeGenerationRouter.PUT("updateDemoCodeGeneration", v1.UpdateDemoCodeGeneration)    // 更新DemoCodeGeneration
		DemoCodeGenerationRouter.GET("findDemoCodeGeneration", v1.FindDemoCodeGeneration)        // 根据ID获取DemoCodeGeneration
		DemoCodeGenerationRouter.GET("getDemoCodeGenerationList", v1.GetDemoCodeGenerationList)  // 获取DemoCodeGeneration列表
	}
}
